chromadb/ingest/__init__.py

```diff
@@ -70,8 +70,14 @@ def max_batch_size(self) -> int:
         to submit_embeddings."""
         pass
 
+    @abstractmethod
+    def optimize(self, topic: str) -> SeqId:
+        """Optimize the WAL be pruning unnecessary records and return the number of
+        records pruned. This operation may be slow and should be used sparingly."""
+        pass
+
 
-ConsumerCallbackFn = Callable[[Sequence[EmbeddingRecord]], None]
+ConsumerCallbackFn = Callable[[Sequence[EmbeddingRecord]], Optional[SeqId]]
```

Comment:
We don't dynamically update the max_seq_id in EmbeddingQueue, so this return helps optimize how much of the data has been committed by the segment and allows it to clean up the WAL for a newly created collection with many millions of embeddings, which otherwise would have required Chroma server restart.

Reply:
This is not the case for distributed, as the max_seq_id is not owned by the WAL queue but rather by the underlying segment. For distributed message TTLs might be a better fit for automatic optimization.